core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -584,16 +584,21 @@ public void runInContext() {
 
     @Override
     public void closed(final Status status) {
+      // For cancellations, promptly inform any users of the context that their work should be
+      // aborted. Otherwise, we can wait until pending work is done.
+      if (!status.isOk()) {
+        context.cancel(status.getCause());
+      }
+      // The contract of this.listener is that it's always accessed from inside callExecutor.
```

Comment:
I'm repeating the comment found higher up in this class:
```
// Only accessed from callExecutor.
private ServerStreamListener listener;
```

I hit a unit test case where I didn't serialize on the executor and following happened:
- ServerImpl#streamCreated runs
  - It creates a JumpToApplicationThreadServerStreamListener whose listener is initially null
  - It enqueues a ContextRunnable on callExecutor responsible for wiring up the method from the registry, but it has not yet run
- Now the unit test attempts to cancel the stream. Because the operation is not serialized on callExecutor, we find that JumpToApplicationThreadServerStreamListener.listener is still null, and raise an error.